app/assets/v2/js/activity.js

```diff
@@ -303,53 +303,95 @@ $(document).ready(function() {
-
-      view_comments($parent, allow_close_comment_container, success_callback, override_hide_comments);
-    }).done(function() {
-      // pass
-    })
-      .fail(function() {
-        $parent.parents('.activity.box').find('.error').removeClass('hidden');
-      })
-      .always(function() {
+        view_comments($parent, allow_close_comment_container, success_callback);
+      }).always(function() {
         $parent.parents('.activity.box').find('.loading').addClass('hidden');
         $parent.parents('.box').find('.comment_container textarea').prop('disabled', false);
-        $('.post_comment').prop('disabled', false);
+        $('.editableComment').prop('disabled', false);
+        $(`.comment_row[data-id="${comment_id}"] .comment_options`).prop('disabled', false).removeClass('hidden');
+        $(`.comment_row[data-id="${comment_id}"]`).parent().closest('.box')
+          .find('#container-post_comment')
+          .find('*').prop('disabled', false).removeClass('hidden');
+        // TODO: automatically cancel editing when another edit_comment button is clicked
+        $('.edit_comment').prop('disabled', false);
       });
+    } else {
+      // user input
+      var comment = $parent.parents('.box').find('.comment_container textarea').val();
+
+      $parent.parents('.box').find('.comment_container textarea').prop('disabled', true);
+      $('.post_comment').prop('disabled', true);
+
+      // validation
+      if (!comment) {
+        return;
+      }
+
+      $parent.parents('.activity.box').find('.loading').removeClass('hidden');
+      var has_hidden_comments = $parent.parents('.activity.box').find('.row.comment_row.hidden').length;
+      // increment number
+      var num = $parent.find('span.num').html();
+
+      num = parseInt(num) + 1;
+      $parent.find('span.num').html(num);
+
+      // remote post
+      var params = {
+        'method': 'comment',
+        'comment': comment,
+        'csrfmiddlewaretoken': $('input[name=csrfmiddlewaretoken]').val()
+      };
+      var url = '/api/v0.1/activity/' + $parent.data('pk');
+
+      $.post(url, params, function(response) {
+        var success_callback = function($parent) {
+          $parent.parents('.box').find('.comment_container textarea').val('');
+          $parent.find('textarea').focus();
+
+        };
+        var override_hide_comments = !has_hidden_comments;
+
+        view_comments($parent, allow_close_comment_container, success_callback, override_hide_comments);
+      }).done(function() {
+        // pass
+      })
+        .fail(function() {
+          $parent.parents('.activity.box').find('.error').removeClass('hidden');
+        })
+        .always(function() {
+          $parent.parents('.activity.box').find('.loading').addClass('hidden');
+          $parent.parents('.box').find('.comment_container textarea').prop('disabled', false);
+          $('.post_comment').prop('disabled', false);
+        });
```

Comment:
^de-indent by 1

Reply:
Are you sure? Travis's style check suggested that be indented in the first place.

Reply:
de-indented